include/zephyr/drivers/mipi_dbi.h

```diff
@@ -293,6 +320,45 @@ static inline int mipi_dbi_release(const struct device *dev,
 	return api->release(dev, config);
 }
 
+/**
+ * @brief Configures MIPI DBI tearing effect signal
+ *
+ * Many displays provide a tearing effect signal, which can be configured
+ * to pulse at each vsync interval or each hsync interval. This signal can be
+ * used by the MCU to determine when to transmit a new frame so that the
+ * read pointer of the display never overlaps with the write pointer from the
+ * MCU. This function configures the MIPI DBI controller to delay transmitting
+ * display frames until the selected tearing effect signal edge occurs.
+ *
+ * The delay will occur on the on each call to @ref mipi_dbi_write_display
+ * where the ``frame_incomplete`` flag was set within the buffer descriptor
+ * provided with the prior call, as this indicates the buffer being written
+ * in this call is the first buffer of a new frame.
+ *
+ * Note that most display controllers will need to enable the TE signal
+ * using vendor specific commands before the MIPI DBI controller can react
+ * to it.
+ *
+ * @param dev mipi dbi controller
+ * @param edge which edge of the TE signal to start transmitting on
+ * @param delay_us how many microseconds after TE edge to start transmission
```

Comment:
Some controllers also support TE source (DSI link or External pin). How would you propose to configure this ?

Reply:
Do you have an example I could look at? Not sure I follow here. Do you mean that the MIPI controller supports selecting the TE source between an on chip DSI peripheral and external pin?

Reply:
Yes, supported in https://www.st.com/resource/en/reference_manual/rm0399-stm32h745755-and-stm32h747757-advanced-armbased-32bit-mcus-stmicroelectronics.pdf.
See 34.16.1
![Screenshot from 2024-11-26 16-25-28](https://github.com/user-attachments/assets/8b7c9370-221e-4829-87ae-29e8b9a3d97c)

Reply:
Thanks for pointing this out- one thing to note, this is a _DSI_ controller, not a _DBI_ controller- so it wouldn't fit within the MIPI DBI API itself.

Looking at this more, I do think we'd need something like this within the MIPI DSI API though- it looks like DSI controllers can either signal TE via a dedicated GPIO pin, or via a D-PHY trigger event on the DSI bus itself.